redisvl/extensions/router/semantic.py

```diff
@@ -596,3 +620,200 @@ def to_yaml(self, file_path: str, overwrite: bool = True) -> None:
         with open(fp, "w") as f:
             yaml_data = self.to_dict()
             yaml.dump(yaml_data, f, sort_keys=False)
+
+    # reference methods
+
+    @classmethod
+    def add_route_references(
```

Comment:
Only question is an ergonomic one.. what would be the benefit of using `add_route_references(...)` on a non-instantiated router class, wherein you will still end up passing through the args here to essentially init the class? Isn't it effectively the same as calling `router = SemanticRouter(...); router.add_route_references(...)'? 

I know this is/was one of the tricky and sticky points, so just curious to hear what you're thinking! :)

Reply:
Ahh yeah, I see what you mean! 

Core issue is that class init is pretty rigid and expects certain inputs. It was probably a gap in the initial interface design for sure on my part.